### ISSUE TITLE
Validator wrongly doesn't validate parent property

### ISSUE THREAD

**Issue by romeokienzler** [elyra-ai/elyra#3019]
**Describe the issue**
When selecting a parent output as input in the properties sidebar of a kfp component the validator wrongly doesn't validate

**To Reproduce**
1. clone repo at given commit and examine [this](https://github.com/claimed-framework/component-library/blob/ca9285894f287e58b688f6639ff0ec81165b22e6/pipelines/gharchive_kfp.pipeline) pipeline 

**Screenshots or log output**
![image](https://user-images.githubusercontent.com/5694071/202382391-28e7813b-1335-443e-af1b-825be15589d9.png)

**Expected behavior**
Pipeline validates

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: v3.12.0

**Comment by kiersten-stokes** [comment]
> @kiersten-stokes @ajbozarth what are your thoughts? Was our intention to have a constraint that one can only select an upstream node output as input if the passing mode is `inputPath`? Irrespective of what our intention was, the current behavior appears to be a bug.

Yes, that is intentional behavior. It would be fairly easy to remove that constraint on the backend.
